### PR TITLE
Replication of ResponsiveAppBar from MaterialUI Docs

### PR DESCRIPTION
**Pull request by mahir118**

### CODE REVIEW

**Comment by lugenx** [comment]
@mahir118 thank you for contributing to the project. 

- could you please comment on the related issue, so i can assign it to you? 

this needed to be done before starting to work on the project. so please consider for the next time, and check our [contribute](https://github.com/lugenx/ecohabit/blob/main/README.md#contribute) section for reference as well.

meanwhile, i will request @rcamach7 to review this PR.

thanks again for your interest.

**Comment by mahir118** [comment]
Thanks a lot for letting me know @lugenx. I just commented on the issue.

**Comment by lugenx** [comment]
@Rishav1707 this is the pr that already raised before assignment. you are welcome to check this pr, and give some feedback.

**Comment by Rishav1707** [comment]
> @Rishav1707 this is the pr that already raised before assignment. you are welcome to check this pr, and give some feedback.

But I am already in on this issue #124

**Comment by lugenx** [comment]
> > @Rishav1707 this is the pr that already raised before assignment. you are welcome to check this pr, and give some feedback.
> 
> But I am already in on this issue #124

i understand. i assigned this way, because i didn't want to him to lose his effort, since he is already worked on it. next time i will make sure to assign an issue whoever comments on it first. 

because you commented on the issue first, I can/should still assign it to you and cancel this pr. please let me know.

**Comment by Rishav1707** [comment]
> > > @Rishav1707 this is the pr that already raised before assignment. you are welcome to check this pr, and give some feedback.
> > 
> > 
> > But I am already in on this issue #124
> 
> i understand. i assigned this way, because i didn't want to him to lose his effort, since he is already worked on it. next time i will make sure to assign an issue whoever comments on it first.
> 
> because you commented on the issue first, I can/should still assign it to you and cancel this pr. please let me know.

No, don't cancel his PR.
Let him commit to this issue.
I will work on some other issues next time.

**Comment by rcamach7** [comment]
Hi @mahir118 , thank you for taking on this issue. 

Few things:
* if you're using VSCode, install prettier, and under settings, set it to auto format on save
* Looks like you copied over the Typescript version of the demo app bar, instead of the javascript version. There's a tiny toggle on the upper left corner that lets you switch in between them

Let's take care of that and we'll go from there

**Comment by rcamach7** [comment]
@mahir118 let me know if you need any help on the above

**Comment by mahir118** [comment]
Hi @rcamach7. Sorry for taking so long, I didn't get notified of the texts here. I just made the changes as you said and pushed them. Please let me know of the next steps.

**Comment by rcamach7** [comment]
> Hi @rcamach7. Sorry for taking so long, I didn't get notified of the texts here. I just made the changes as you said and pushed them. Please let me know of the next steps.

Can you run `git log --oneline` and paste the output here?

**Comment by mahir118** [comment]
Sure, here is what I got after running git log --oneline:

- 9d671e7 (HEAD -> added_ResponsiveAppBar, origin/added_ResponsiveAppBar) I have made the required changes
- 92ffbbd (origin/main, origin/HEAD, main) Merge pull request #112 from Emelit0/authentication
- 7d195e8 Merge pull request #121 from amiranga/update_contribute_sec_in_readme
- c88daa7 Merge branch 'update_contribute_sec_in_readme' of github.com:amiranga/ecohabit into update_contribute_sec_in_readme
- 5fd7c44 Updated guidelines in Contribute section in README
- 8bab412 Merge pull request #115 from Rishav1707/contribute
- d0afc97 Update CONTRIBUTING.md
- 60d11b7 Updated guidelines in Contribute section in README
- ce31e51 Merge pull request #117 from Rishav1707/socials
- 81c6b1d Merge pull request #116 from amiranga/example_env_file_for_server
- :...skipping...
- 9d671e7 (HEAD -> added_ResponsiveAppBar, origin/added_ResponsiveAppBar) I have made the required changes
- 92ffbbd (origin/main, origin/HEAD, main) Merge pull request #112 from Emelit0/authentication
- 7d195e8 Merge pull request #121 from amiranga/update_contribute_sec_in_readme
- c88daa7 Merge branch 'update_contribute_sec_in_readme' of github.com:amiranga/ecohabit into update_contribute_sec_in_readme
- 5fd7c44 Updated guidelines in Contribute section in README
- 8bab412 Merge pull request #115 from Rishav1707/contribute
- d0afc97 Update CONTRIBUTING.md
- 60d11b7 Updated guidelines in Contribute section in README
- ce31e51 Merge pull request #117 from Rishav1707/socials
- 81c6b1d Merge pull request #116 from amiranga/example_env_file_for_server
- 6de9a74 Addressed PR comments
- 9d6ce24 Update README.md
- af309e9 Readme file updated
- 4ae6031 Merge pull request #114 from SoyebSarkar/newBranch
- 0d6a142 Readme file modified
- 963d5a5 fixed typos
- 902504d remove middleware on routes and added middleware on index
- bc4f993 no password on response
- bb2cb49 Added .env.example file for server
- 9859689 Renamed how-to-contribute.md file
- 689814d Corrected cors origin domain name to .org from .com in server/index.js file
- b74243e add middleware for password validation
- 7c7388f Merge pull request #106 from xinyi-dev/102-refactor-export-and-import
- 4583b50 Merge pull request #110 from lugenx/109-add-discord-link

**Comment by rcamach7** [comment]
@mahir118 It appears there is no record of new changes, both on your computer, or on github. 

This means you need to redo all the new changes in this same branch, stage them, and commit them. 
After you do this, run the same line, `git log --oneline`, and you should see a new commit that you will have just made. 

Afterwards push the changes, and we will see them here.

**Comment by mahir118** [comment]
@rcamach7 Sure, I just pushed the changes. Thanks for letting me know.

**Comment by lugenx** [comment]
@mahir118, could you please edit the name of the pull request and make it more specific? please make it similar to the title of the issue you are addressing.

**Comment by rcamach7** [comment]
> @rcamach7 Sure, I just pushed the changes. Thanks for letting me know.

It looks like you don't have prettier installed and set to format on save. To do this:
1) Install the Prettier extension for VSCode.
2) Open the VSCode settings, and search for prettier
3) In the "Editor: Format On Save" setting, make sure it is turned on.
4) Open the files you have worked on hit save.

File: client/src/components/ResponsiveAppBar/ResponsiveAppBar.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import AppBar from '@mui/material/AppBar';
```
Comment (rcamach7) [inline client/src/components/ResponsiveAppBar/ResponsiveAppBar.js:2]: These are repeated imports that will be removed once you have imported the components and icons separately
- TODO: Remove imports from line 6 to 18 once you have imported the components and icons separately

File: client/src/components/ResponsiveAppBar/ResponsiveAppBar.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import MenuIcon from '@mui/icons-material/Menu';
+import Container from '@mui/material/Container';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import Tooltip from '@mui/material/Tooltip';
+import MenuItem from '@mui/material/MenuItem';
+import AdbIcon from '@mui/icons-material/Adb';
+
```
Comment (rcamach7) [inline client/src/components/ResponsiveAppBar/ResponsiveAppBar.js:15]: You should have two separate imports, one for the icons and one for the components (see where "from" is coming from)
- TODO: Remove any icon imports from the line blow, and import them separately (see below)

File: client/src/components/ResponsiveAppBar/ResponsiveAppBar.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import { AppBar, Box, Toolbar, IconButton, Typography, Menu, MenuIcon, Container, Avatar, Button, Tooltip, MenuItem, AdbIcon} from '@mui/material';
+
```
Comment (rcamach7) [inline client/src/components/ResponsiveAppBar/ResponsiveAppBar.js:17]: If you look at the original imports, you'll notice the icon name is actually "Adb" from MUI, so we will rename the import to AdbIcon. Import all icons like so: 
- TODO: Import all icons in the line below, and rename them to their respective icon name

```js
import { Adb as AdbIcon } from '@mui/icons-material';
```

File: client/src/components/ResponsiveAppBar/ResponsiveAppBar.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import Menu from '@mui/material/Menu';
+import MenuIcon from '@mui/icons-material/Menu';
+import Container from '@mui/material/Container';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import Tooltip from '@mui/material/Tooltip';
+import MenuItem from '@mui/material/MenuItem';
+import AdbIcon from '@mui/icons-material/Adb';
+
+import { AppBar, Box, Toolbar, IconButton, Typography, Menu, MenuIcon, Container, Avatar, Button, Tooltip, MenuItem, AdbIcon} from '@mui/material';
```
Comment (rcamach7) [inline client/src/components/ResponsiveAppBar/ResponsiveAppBar.js:16]: You should have two separate imports, one for the icons and one for the components (see where "from" is coming from)
- TODO: Remove any icon imports from the line above, and import them separately (see below)

File: client/src/components/ResponsiveAppBar/ResponsiveAppBar.js (hunk 1 above)
Comment (rcamach7) [inline client/src/components/ResponsiveAppBar/ResponsiveAppBar.js:2]: These are repeated imports that will be removed once you have imported the components and icons separately
- TODO: Remove imports from line 2 to 14 once you have imported the components and icons separately

**Comment by mahir118** [comment]
Thank you so much for your help @lugenx and @rcamach7. I have made the above changes. Can you please let me know how can I change the **name of the pull request** and **push my new changes**?

**Comment by lugenx** [comment]
> Thank you so much for your help @lugenx and @rcamach7. I have made the above changes. Can you please let me know how can I change the **name of the pull request** and **push my new changes**?

@mahir118 I just renamed the PR for you. please push your changes to the same `added_ResponsiveAppBar` branch.

**Comment by mahir118** [comment]
Sure @lugenx. I just pushed the changes.

**Comment by rcamach7** [comment]
Thank you @mahir118 , looking good so far! The last part of the ticket would be: 

* Replace the old navbar with the new one in `client/src/components/MainLayout/MainLayout.js`, and delete the old import
* run `npm start` inside the `client` directory and confirm the new navbar is being rendered

Let me know if you have any questions!

**Comment by mahir118** [comment]
Sure, I will do that @rcamach7. However, when you say replace, should I replace **MainLayout.js** (client/src/components/MainLayout/MainLayout.js) with **ResponsiveAppBar.js** (client/src/components/ResponsiveAppBar/ResponsiveAppBar.js) and then delete ResponsiveAppBar.js from client/src/components/ResponsiveAppBar ? Also, can you please guide me a little on how to run **npm start** inside the **client** directory?

File: client/src/components/MainLayout/MainLayout.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { Outlet } from "react-router-dom";
 
 import NavBar from "../NavBar/NavBar";
 import Footer from "../Footer/Footer";
+import ResponsiveAppBar from "../ResponsiveAppBar/ResponsiveAppBar";
```
Comment (rcamach7) [inline client/src/components/MainLayout/MainLayout.js:6]: In this file, you imported the new ResponsiveAppBar. The main component in this file is still using `Navbar`. 
- In line 15 of this file, you will be replacing `NavBar` with `ResponsiveAppBar`
- then delete the import of `NavBar` in line 4.

To run the app using npm start first: 
- open the integrated terminal in VsCode
- run `cd client` on the terminal to change into the client folder
- run `npm start` to start the app and launch it on browser
You should now see the new app navbar in the browser. Let me know if you have any issues

Comment (mahir118) [reply client/src/components/MainLayout/MainLayout.js:6]: Great, just did that. However, **npm start** gave me the following output:

            > ecohabit@0.1.0 start
            > react-scripts start
            sh: react-scripts: command not found

I looked it up on [stack overflow](https://stackoverflow.com/questions/47612580/react-scripts-command-not-found) and found out there might be issues with packages installation. What would be your recommendation?

Comment (lugenx) [reply client/src/components/MainLayout/MainLayout.js:6]: @mahir118, do you see a folder named "node_modules" in the "client" folder? if not, please use `npm install` and wait for the installation complete. after that try `npm start` again.

Comment (mahir118) [reply client/src/components/MainLayout/MainLayout.js:6]: Got it, it's working perfectly now. Thanks a lot for all your help. Is there anything else I can do from this point?

**Review by lugenx — APPROVED** [review]

**Comment by lugenx** [comment]
thank you so much @mahir118

**Pull request merged**